### ISSUE TITLE
Run Sidekiq instance as only a server or only as a client

### ISSUE THREAD

**Issue by SemmyTan** [mperham/sidekiq#3940]
Hello,

Is there a way to configure Sidekiq to only act as a client or only as a server? I basically want an instance of Sidekiq to either only push or pull jobs from Redis, but not both. I am trying to run multiple remote Sidekiq servers for load distribution and high availability so each Sidekiq instance will act as only a client or only a server but they will all connect to the same Redis instance. I searched through the documentation and Google but wasn't able to find a way to do this, so maybe my search skills are lacking.

I know that the Redis options can be configured separately for the server and client with `Sidekiq.configure_server` and `Sidekiq.configure_client`. I tried setting the Redis url to `nil`, but that just defaulted to localhost. I also tried setting the url to a path where Redis was not running, but Sidekiq raised an error for `Error connecting to Redis` and wouldn't start.

I know as a workaround I can point the Redis URL to an actual valid Redis instance that is always empty, but I would prefer not running another process/server just for that purpose if possible.

Are there any options like this that I am missing?
``` ruby
# Enable client
Sidekiq.configure_client do |config|
  config.redis = { url: 'redis://validredishost.com:6379' }
end

# But disable server
Sidekiq.configure_server do |config|
  config.redis = { disabled: true } # Something like this
  config.disabled = true # Or this
end
```

Currently using Ruby 2.2.3 / Sidekiq 5.0.5 / Enterprise 1.6.1, but can update to the latest stable versions if needed.

Thanks,
Semmy

**Comment by mperham** [comment]
A Sidekiq process is always “server”. Anything not Sidekiq (puma, rake, etc) is a client.

**Comment by SemmyTan** [comment]
@mperham Thank you for the fast response. After reading your response, I looked over everything again and the problem I encountered with my testing was due to my development start script starting up Sidekiq each time, which was why the jobs were still being processed on the client. The solution for this is like you said: only start up Sidekiq with `bundle exec sidekiq` on the server instances and not start it on the client instances. Just adding the Sidekiq gem alone on the client instances allow them to push the jobs to Redis.

Sorry for taking up your time and thanks again!
